src/components/checkbox/checkbox.js

```diff
@@ -169,19 +169,29 @@ function MdCheckboxDirective(inputDirective, $mdAria, $mdConstant, $mdTheming, $
         }
       }
 
+      /**
+       * @param {KeyboardEvent} ev 'keypress' event to handle
+       */
       function keypressHandler(ev) {
         var keyCode = ev.which || ev.keyCode;
+        var submit, form;
+
         ev.preventDefault();
         switch (keyCode) {
           case $mdConstant.KEY_CODE.SPACE:
             element.addClass('md-focused');
             listener(ev);
             break;
           case $mdConstant.KEY_CODE.ENTER:
-            var form = $mdUtil.getClosest(ev.target, 'form');
-            // We have to use a native event here as the form directive does not use jqlite
+            // Match the behavior of the native input type="checkbox".
+            // It's important that MouseEvent is used here versus Event in order to match the native
+            // behavior and properly trigger ng-submit.
```

Comment:
I think the `Event` part of this comment is no longer necessary. I would, however, expand the comment like:
```ts
// Match the behavior of the native <input type="checkbox">.
// When the enter key is pressed while focusing a native checkbox
// inside a form, the browser will trigger a `click` on the first non-disabled
// submit button in the form. Note that this is different from text inputs, which
// will directly submit the form without needing a submit button to be present.
```

Reply:
Good catch, updated.